src/shared/components/workbasket-filters/workbasket-filters.component.ts

```diff
@@ -84,20 +84,26 @@ export class WorkbasketFiltersComponent implements OnInit {
       queryParams[WorkbasketFiltersComponent.PARAM_CASE_STATE] = this.selected.caseState.id;
     }
     if (init) {
-      this.windowService.setLocalStorage('savedQueryParams', JSON.stringify(queryParams));
+      this.windowService.setLocalStorage(SAVED_QUERY_PARAM_LOC_STORAGE, JSON.stringify(queryParams));
     }
     // without explicitly preserving alerts any message on the page
     // would be cleared out because of this initial navigation.
     // The above is only true if no alerts were set prior to loading case list page.
     if (!this.alertService.isPreserveAlerts()) {
       this.alertService.setPreserveAlerts(!this.initialised);
     }
-    this.selected.formGroup = this.formGroup;
+    if (Object.keys(this.formGroup.controls).length === 0) {
+      this.selected.formGroup  = JSON.parse(localStorage.getItem(FORM_GROUP_VAL_LOC_STORAGE));
+    } else {
+      this.selected.formGroup = this.formGroup;
+    }
     this.selected.init = init;
     this.selected.page = 1;
     this.selected.metadataFields = this.getMetadataFields();
     if (init) {
-      this.windowService.setLocalStorage(FORM_GROUP_VAL_LOC_STORAGE, JSON.stringify(this.formGroup.value));
+      if (Object.keys(this.formGroup.controls).length > 0) {
+        this.windowService.setLocalStorage(FORM_GROUP_VAL_LOC_STORAGE, JSON.stringify(this.formGroup.value));
+      }
     }
```

Comment:
can this if(init) {} block merge with line 86 to 88, another if(init) block there.